/*
 * Copyright (C) 2019 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

option java_package = "com.google.android.pwrstatsutil";
package com.google.android.pwrstatsutil;


message PowerStatistic {
    oneof power_stat {
        StateResidency power_entity_state_residency = 1;
        RailEnergy rail_energy = 2;
        StateResidency c_state_residency = 3;
        // add new power_stats here
    }
}

// Utility message for items that provide a state residency in milliseconds
message StateResidency {
    message Residency {
        string entity_name = 1;
        string state_name = 2;
        uint64 time_ms = 3;
    }

    repeated Residency residency = 1;
}

// Rail energy data in uWs
message RailEnergy {
    message RailEntry {
        string rail_name = 1;
        uint64 energy_uws = 2;
    }

    repeated RailEntry entry = 1;
}
